src/js/listeners/post/searchListing.js

```diff
@@ -25,6 +29,47 @@ export async function searchListings() {
     else {
       renderNoListings();
     }
- 
   };
 }
+
+export async function filterListings() {
+
+    const arrayOfListings = await getListOfListings();
+
+    function filter(daysInMilliseconds) {
+      const filteredListings = arrayOfListings.filter((listing) => {
+      const dateCreated = new Date(listing.created);
+      const currentDate = new Date();
+      const timeDifference = currentDate.getTime() - dateCreated.getTime();
+      if (timeDifference <= daysInMilliseconds) {
+        return true
+      }
+     });
+     if (filteredListings.length > 0) {
+      renderListings(filteredListings)
+     } else {
+      renderNoListings()
+     }
+    }
+    
+    const btnAll = document.querySelector('#allListings');
+    const btnLatest = document.querySelector('#latestListings');
+    const btnFourteen = document.querySelector('#fourteenDaysListings');
+    const btnThirty = document.querySelector('#thirtyDaysListings');
+
+    btnAll.addEventListener('click', () => {
+    renderListings(arrayOfListings);
+    } );
+
+    btnLatest.addEventListener('click', () => {
+      filter(4 * 24 * 60 * 60 * 1000)
```

Comment:
I see the time intervals are hardcoded, would a better option be to define them as constants at the start of the function for better read/maintainability?